jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -116,12 +134,212 @@ MlirToJlmConverter::ConvertOperation(
   else if (auto constant = ::mlir::dyn_cast<::mlir::arith::ConstantIntOp>(&mlirOperation))
   {
     auto type = constant.getType();
+    // std::cout << "Casted to ConstantIntOp with t value: " << constant.value() << "\n";
     JLM_ASSERT(type.getTypeID() == ::mlir::IntegerType::getTypeID());
-    auto integerType = ::mlir::cast<::mlir::IntegerType>(type);
+    auto integerType = type.isIntOrFloat() ? type.cast<::mlir::IntegerType>() : nullptr;
```

Comment:
You assert the line above that it is an integer type. The conditional is unnecessary.